awx_collection/plugins/modules/role.py

```diff
@@ -272,12 +270,18 @@ def main():
                     lookup_data_populated = {}
                 else:
                     lookup_data_populated = lookup_data
-            data = module.get_one(key, name_or_id=resource, data=lookup_data_populated)
+                if key == 'target_teams':
+                    lookup_key = 'teams'
+                else:
+                    lookup_key = key
```

Comment:
I don't like the indent of this. If the condition `key in resources` is not met, then `lookup_key` stands to be undefined and then referenced on the `module.get_one` line.